### DISCUSSION TITLE
PAGImageView 播放网络连接资源，释放的时候removeFromSuperview 崩溃

### DISCUSSION THREAD

**Discussion by jate** [Tencent/libpag#1983] [General] [answered]
<img width="1944" alt="截屏2023-12-11 18 10 41" src="https://github.com/Tencent/libpag/assets/3424780/99f6bcc7-ed90-4cc2-a5db-0b9718102962">

播放动效正常，多个实例播放同一个资源，在释放的时候 移除superview崩溃。不知道为啥。

**Comment by jate** [comment]
同时存在错误log：PAGImageView: CVPixelBufferRef create failed!

**Reply by kevingpqi123** [reply]
这类问题建议提供完整的复现工程，可以基于 pag 的官方 demo 构造一个

**Comment by jate** [comment, marked as answer]
崩溃问题解决方案：

先移除约束后，在removeFromSuperView。

<img width="1202" alt="截屏2023-12-12 09 15 05" src="https://github.com/Tencent/libpag/assets/3424780/6c4e9c02-d12c-425b-901a-4010159e2c24">

https://copyprogramming.com/howto/crash-issue-ios-12-0-1-16a404-app-closed-automatically
